cirq-core/cirq/sim/density_matrix_simulator_test.py

```diff
@@ -687,7 +687,7 @@ def test_simulate_sweeps_param_resolver(dtype):
 def test_simulate_moment_steps(dtype):
     q0, q1 = cirq.LineQubit.range(2)
     circuit = cirq.Circuit(cirq.H(q0), cirq.H(q1), cirq.H(q0), cirq.H(q1))
-    simulator = cirq.DensityMatrixSimulator(dtype=dtype)
+    simulator = cirq.DensityMatrixSimulator(dtype=dtype, split_untangled_states=False)
```

Comment:
The fact that this fails with `split_untangled_states=True` (rounding error) indicates that this param should be False by default. The perf optimization should be opt-in.